pkg/process/metadata/parser/java/util.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type mapSource struct {
+	m map[string]string
+}
+
+// Get the value for the supplied key
+func (y *mapSource) Get(key string) (string, bool) {
+	val, ok := y.m[key]
+	return val, ok
+}
+
+// GetDefault gets the value for the supplied key or the defVal if missing
+func (y *mapSource) GetDefault(key string, defVal string) string {
+	val, ok := y.m[key]
+	if !ok {
+		return defVal
+	}
+	return val
+}
+
+func newArgumentSource(arguments []string, prefix string) *mapSource {
+	parsed := make(map[string]string)
+	for _, val := range arguments {
+		if strings.HasPrefix(val, prefix) {
+			parts := strings.SplitN(val[len(prefix):], "=", 2)
+			if len(parts) == 1 {
+				parsed[parts[0]] = ""
+			} else {
+				parsed[parts[0]] = parts[1]
+			}
+			continue
+		}
+	}
+	return &mapSource{parsed}
+}
+
+func newPropertySourceFromStream(rc io.ReadCloser, filename string) (*props.PropertyGetter, error) {
+	var properties props.PropertyGetter
+	var err error
+	if strings.HasSuffix(filename, ".properties") {
+		properties, err = props.Read(rc)
+	} else if strings.HasSuffix(filename, ".yaml") || strings.HasSuffix(filename, ".yml") {
+		properties, err = newYamlSource(rc)
+	} else {
+		err = fmt.Errorf("unhandled file type for %s", filename)
+	}
+	return &properties, err
+}
+
+func longestPathPrefix(pattern string) string {
+	idx := strings.IndexAny(pattern, "?*")
+	if idx < 0 {
+		return pattern
+	}
+	idx = strings.LastIndex(pattern[:idx], "/")
+	if idx < 0 {
+		return ""
+	}
+	return pattern[:idx]
+}
+
+func scanSourcesFromFileSystem(profilePatterns map[string][]string) map[string]*props.Combined {
+	ret := make(map[string]*props.Combined)
+	matcher := antpath.New()
+	for profile, pp := range profilePatterns {
+		for _, pattern := range pp {
+			path := longestPathPrefix(pattern)
+			_ = filepath.WalkDir(filepath.FromSlash(path), func(p string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				path := filepath.ToSlash(p)
+
+				if d.IsDir() {
+					if !matcher.MatchStart(pattern, path) {
+						return filepath.SkipDir
+					}
+				} else if matcher.Match(pattern, path) {
+					// found
+					var err error
```

Comment:
redundant declaration
1. you already have `err` defined in the scope of the function (the argument `err`)
2. you're declaring `err` (again) in the following line